index.js

```diff
@@ -5,87 +5,75 @@
 // NOTE: addPeer should take in an optional second argument, flags
 // TODO: destroy wire if peer sends PEX messages too frequently
 
-var EventEmitter = require('events').EventEmitter
-var compact2string = require('compact2string')
-var string2compact = require('string2compact')
-var bencode = require('bencode')
-var inherits = require('inherits')
+const { EventEmitter } = require('events')
+const bencode = require('bencode')
+const compact2string = require('compact2string')
+const string2compact = require('string2compact')
 
-var PEX_INTERVAL = 65000 // just over one minute
-var PEX_MAX_PEERS = 50 // max number of peers to advertise per PEX message
+const PEX_INTERVAL = 65000 // just over one minute
+const PEX_MAX_PEERS = 50 // max number of peers to advertise per PEX message
 
-module.exports = function () {
-  inherits(utPex, EventEmitter)
+class utPex extends EventEmitter {
+  constructor (wire) {
+    super()
 
-  function utPex (wire) {
-    var self = this
-    EventEmitter.call(self)
+    this._wire = wire
+    this._intervalId = null
 
-    self._wire = wire
-    self._intervalId = null
-
-    self.reset()
+    this.reset()
   }
 
-  utPex.prototype.name = 'ut_pex'
-
   /**
    * Start sending regular PEX updates to remote peer.
    */
-  utPex.prototype.start = function () {
-    var self = this
-    clearInterval(self._intervalId)
-    self._intervalId = setInterval(self._sendMessage.bind(self), PEX_INTERVAL)
-    if (self._intervalId.unref) self._intervalId.unref()
+  start () {
+    clearInterval(this._intervalId)
+    this._intervalId = setInterval(this._sendMessage.bind(this), PEX_INTERVAL)
```

Comment:
I think you could wrap this in an arrow function instead of using `bind()`

```js
setInterval(() => { this._sendMessage() }, PEX_INTERVAL)
```

Reply:
will change that